### ISSUE TITLE
Stop logback from shutdown hook

### ISSUE THREAD

**Issue by patriknw** [akka/akka#28212]
Might be important for flushing/closing async appenders.

If shutdown is added with [logback.xml configuration](http://logback.qos.ch/manual/configuration.html#stopContext) I suspect that it may run before ActorSystem shutdown since order of jvm hooks are undefined.

Since there might be several ActorSystems in same JVM we can't just do it after AS termination.

**Comment by TimMoore** [comment]
Attn: @marcospereira @ignasi35 this might impact Play, too.

Play [disables Akka's shutdown hook](https://github.com/playframework/playframework/blob/bd87d8c4d742c5ffc6c940ccc39ee0d7ce08373b/core/play/src/main/resources/play/reference-overrides.conf#L26), and it stops the logging framework from a [`CoordinatedShutdown.PhaseBeforeActorSystemTerminate` task](https://github.com/playframework/playframework/blob/bd87d8c4d742c5ffc6c940ccc39ee0d7ce08373b/transport/server/play-akka-http-server/src/main/scala/play/core/server/AkkaHttpServer.scala#L487). That might be too early, if the `ActorSystem` logs anything while terminating.

**Comment by patriknw** [comment]
> stops the logging framework from a CoordinatedShutdown.PhaseBeforeActorSystemTerminate task. That might be too early, if the ActorSystem logs anything while terminating.

yes, that might be too early. Might be better via `ActorSystem.whenTerminated` or the `Future` returned by `CoordinatedShutdown.run`.

For Akka itself we probably also have to consider that that logback can be shared by several ActorSystems in same JVM.

**Comment by ignasi35** [comment]
> Play [disables Akka's shutdown hook](https://github.com/playframework/playframework/blob/bd87d8c4d742c5ffc6c940ccc39ee0d7ce08373b/core/play/src/main/resources/play/reference-overrides.conf#L26), and it stops the logging framework from a [`CoordinatedShutdown.PhaseBeforeActorSystemTerminate` task](https://github.com/playframework/playframework/blob/bd87d8c4d742c5ffc6c940ccc39ee0d7ce08373b/transport/server/play-akka-http-server/src/main/scala/play/core/server/AkkaHttpServer.scala#L487). That might be too early, if the `ActorSystem` logs anything while terminating.

That code requires a review. The shutdown task should be renamed and the Logger should be stopped from somewhere else.

(better continue this discussion on an issue in [playframework/playframework/](https://github.com/playframework/playframework/).
